### PR TITLE
[SYCL] Add one more optional kernel features test

### PR DESCRIPTION
**Pull request by AlexeySachkov**

### CODE REVIEW

**Comment by AlexeySachkov** [comment]
The only failed test is the following:
```
[2023-02-08T23:56:03.174Z] Failed Tests (1):
[2023-02-08T23:56:03.174Z]   SYCL :: ESIMD/imulh_umulh.cpp
```

It is unrelated to this PR. Upd: according to https://github.com/intel/llvm-test-suite/pull/1566, it should disappear. Since https://github.com/intel/llvm/pull/8176 is already merged, that will probably happen after the next nightly

File: SYCL/OptionalKernelFeatures/no-speculative-compilation.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+//
+// This test is indented to check that no speculative compilation is happenning,
```
Comment (steffenlarsen) [inline SYCL/OptionalKernelFeatures/no-speculative-compilation.cpp]:
```suggestion
// This test intends to check that no speculative compilation is happening,
```

**Review by steffenlarsen — APPROVED** [review]
Minor correction, but otherwise LGTM!

**Comment by AlexeySachkov** [comment]
New change only modifies comment, so it should be ok to merge it without waiting for another for pre-commit to complete

**Pull request merged**